src/V3Coverage.cpp

```diff
@@ -542,6 +577,230 @@ class CoverageVisitor final : public VNVisitor {
         lineTrack(nodep);
     }
 
+    void finalizeExprComments() {
+        for (CoverExpr& expr : m_exprs) {
+            UASSERT(m_seeking != NONE, "Finalizing expression comments without an objective");
+            expr.m_comment = "(" + expr.m_comment + ") => " + (m_objective ? "1" : "0");
+        }
+    }
+
+    bool checkMaxExprs() {
+        if (m_seeking == ABORTED) { return true; }
+        if (static_cast<int>(m_exprs.size()) <= v3Global.opt.coverageExprMax()) { return false; }
+
+        m_seeking = ABORTED;
+        for (CoverExpr& expr : m_exprs) { expr.m_expr->deleteTree(); }
```

Comment:
Rather than deleting now use pushDeletesp - easier to debug if nodes don't disappear midvisitor.